src/daft-sql/src/modules/aggs.rs

```diff
@@ -16,28 +17,27 @@ pub struct SQLModuleAggs;
 
 impl SQLModule for SQLModuleAggs {
     fn register(parent: &mut SQLFunctions) {
-        use AggExpr::{Count, Max, Mean, Min, Stddev, Sum};
         // HACK TO USE AggExpr as an enum rather than a
         let nil = Arc::new(Expr::Literal(LiteralValue::Null));
-        parent.add_fn(
-            "count",
-            Count(nil.clone(), daft_core::count_mode::CountMode::Valid),
-        );
-        parent.add_fn("sum", Sum(nil.clone()));
-        parent.add_fn("avg", Mean(nil.clone()));
-        parent.add_fn("mean", Mean(nil.clone()));
-        parent.add_fn("min", Min(nil.clone()));
-        parent.add_fn("max", Max(nil.clone()));
-        parent.add_fn("stddev", Stddev(nil.clone()));
-        parent.add_fn("stddev_samp", Stddev(nil));
+        parent.add_fn("count", AggExpr::Count(nil.clone(), CountMode::Valid));
+        parent.add_fn("count_distinct", AggExpr::CountDistinct(nil.clone()));
+        parent.add_fn("sum", AggExpr::Sum(nil.clone()));
+        parent.add_fn("avg", AggExpr::Mean(nil.clone()));
+        parent.add_fn("mean", AggExpr::Mean(nil.clone()));
+        parent.add_fn("min", AggExpr::Min(nil.clone()));
+        parent.add_fn("max", AggExpr::Max(nil.clone()));
+        parent.add_fn("stddev", AggExpr::Stddev(nil.clone()));
+        parent.add_fn("stddev_samp", AggExpr::Stddev(nil));
     }
 }
 
 impl SQLFunction for AggExpr {
     fn to_expr(&self, inputs: &[FunctionArg], planner: &SQLPlanner) -> SQLPlannerResult<ExprRef> {
         // COUNT(*) needs a bit of extra handling, so we process that outside of `to_expr`
         if let Self::Count(_, _) = self {
-            handle_count(inputs, planner)
+            handle_count(inputs, planner, DuplicateTreatment::All)
+        } else if let Self::CountDistinct(_) = self {
+            handle_count(inputs, planner, DuplicateTreatment::Distinct)
```

Comment:
Since you are handling the distinction between `count` and `count_distinct` in `SQLPlanner.plan_function`, could you just let count distinct go through the else branch below? As far as I know, `COUNT(DISTINCT *)` is not a thing so we don't have to handle the special count case that `handle_count` is used for.

Reply:
This makes sense. Updated.